packages/create-pantheon-decoupled-kit/src/templates/next-drupal-umami/pages/[...alias].jsx

```diff
@@ -0,0 +1,189 @@
+import { getCurrentLocaleStore, globalDrupalStateStores } from '../lib/stores';
+import { getPreview } from '../lib/getPreview';
+import { translatePath } from '@pantheon-systems/drupal-kit';
+import { NextSeo } from 'next-seo';
+import { IMAGE_URL } from '../lib/constants';
+
+import Link from 'next/link';
+import Layout from '../components/layout';
+import { Recipe } from '@pantheon-systems/nextjs-kit';
+import { ContentWithImage } from '@pantheon-systems/nextjs-kit';
+
+export default function CatchAllRoute({
+	pageData,
+	hrefLang,
+	footerMenu,
+	preview,
+}) {
+	const RenderPage = () => {
+		if (pageData?.type === 'node--page') {
+			const {
+				title,
+				body: { processed },
+			} = pageData;
+			return (
+				<>
+					<article className="prose lg:prose-xl mt-10 mx-auto">
+						<h1>{title}</h1>
+						<Link passHref href="/pages" className="font-normal">
+							Pages &rarr;
+						</Link>
+						<div className="mt-12 max-w-lg mx-auto lg:grid-cols-3 lg:max-w-screen-lg">
+							<div dangerouslySetInnerHTML={{ __html: processed }} />
+						</div>
+					</article>
+				</>
+			);
+		}
+
+		if (pageData?.type === 'node--article') {
+			const {
+				title,
+				body: { processed },
+				field_media_image,
+				thumbnail,
+			} = pageData;
+			const imgSrc = field_media_image?.field_media_image?.uri.url;
+
+			return (
+				<ContentWithImage
+					title={title}
+					content={processed}
+					imageProps={
+						imgSrc
+							? {
+									src: IMAGE_URL + imgSrc,
+									alt: thumbnail?.resourceIdObjMeta?.alt,
+							  }
+							: undefined
+					}
+				/>
+			);
+		}
+
+		if (pageData.type === 'node--recipe') {
+			const {
+				title,
+				field_recipe_category,
+				field_media_image,
+				field_ingredients,
+				field_recipe_instruction,
+				thumbnail,
+			} = pageData;
+
+			const imgSrc = field_media_image?.field_media_image?.uri.url;
+			return (
+				<Recipe
+					title={title}
+					category={field_recipe_category[0].name}
+					imageProps={
+						imgSrc
+							? {
+									src: IMAGE_URL + imgSrc,
+									alt: thumbnail?.resourceIdObjMeta?.alt,
+							  }
+							: undefined
+					}
+					ingredients={field_ingredients}
+					instructions={field_recipe_instruction.value}
+				/>
+			);
+		}
+		return (
+			<>
+				<h2 className="text-xl text-center mt-14">No content found 🏜</h2>
+			</>
+		);
+	};
```

Comment:
@CobyPear I see how this could reduce the number of individual template files, however I do have some implementation concerns with this.

If some umami data is wrapped in a handlebars conditional inside of the next-drupal template, wouldn't that defeat the purpose of this addon? With that approach, in order to add umami data into an  preexisting project, a user would need to run both generators to get all of the umami data. Rerunning both of the generators would also overwrite any changes someone would have made to their project.

I can see this working if we are only giving the option to add umami data during the initial project generation but otherwise it makes the addon harder and more confusing to use IMO.

Thoughts?

Reply:
> With that approach, in order to add umami data into an preexisting project, a user would need to run both generators to get all of the umami data.

I don't see why that would be true. For example, if you configure the generator to pass in `umami: true` to the `data` in the actions, the templates can render only the pieces in the `umami` conditional.

>  overwrite any changes someone would have made to their project.

This is why we have the diff and prompt on each change.